ethcore/sync/src/chain/mod.rs

```diff
@@ -648,8 +656,28 @@ impl ChainSync {
 		}
 	}
 
+	/// Check if the snapshot service is ready
+	fn is_snapshot_ready(&mut self, io: &SyncIo) -> bool {
+		if io.snapshot_service().ready() {
+			trace!(target: "snapshot", "Snapshot Service is ready!");
+			// Sync the previously resumed chunks
+			self.snapshot.sync(io);
```

Comment:
Calling that method multiple times will overwrite `completed_chunks`, I don't think it's expected.

Reply:
Yep but always with the right value. Is that an issue?

Reply:
Are we sure it's the right value? `completed_chunks` are modified in `snapshot`, no? Are we super sure this two are in sync?

If it's required only initially, maybe change the type of `completed_chunks` and only restore if it's `None`?